src/odemis/driver/test/simcam_test.py

```diff
@@ -302,29 +302,54 @@ def test_focus(self):
         f.result()
         self.assertEqual(self.focus.position.value, pos)
 
-    def test_get_center(self):
+
+class TestSimCamMove(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # Full image is 912x912, so use about one third, to have space to move around
+        cls.camera = CLASS(max_res=(300, 350), **KWARGS)
+        cls.camera.updateMetadata({model.MD_PIXEL_SIZE: (1e-6, 1e-6)})
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.camera.terminate()
+
+    @timeout(20)
+    def test_move_around(self):
         """
-        Test getCenter function
+        Test that moving the "stage" moves the image
         """
-        # Test without cropping simulated image
-        shape = self.camera.shape
-        center = self.camera._get_center()
-        self.assertEqual(center, (shape[0] / 2, shape[1] / 2))
-        # Test with cropping image
-        cropped_shape = (int(shape[0]/4), int(shape[1]/4))
-        self.camera = CLASS(dependencies={"focus": self.focus}, max_res=cropped_shape, **KWARGS)
-        center = self.camera._get_center()
-        self.assertEqual(center, (cropped_shape[0] / 2, cropped_shape[1] / 2))
-        # Simulate stage move
-        self.camera._metadata[model.MD_POS] = (1e-3, 1e-3)
-        self.camera._metadata[model.MD_PIXEL_SIZE] = (1e-5, 1e-5)
-        center = self.camera._get_center()
-        self.assertEqual(center, (cropped_shape[0] / 2 + 100, cropped_shape[1] / 2 + 100))
-        # Move far away
-        self.camera._metadata[model.MD_POS] = (1e-2, 1e-2)
-        center = self.camera._get_center()
-        self.assertEqual(center, (shape[0] - cropped_shape[0]/2, shape[1] - cropped_shape[1]/2))
+        # Test without MD_POS
+        im = self.camera.data.get()
+
+        # Now, at 0,0 (should be the same)
+        self.camera.updateMetadata({model.MD_POS: (0, 0)})
+        im0 = self.camera.data.get()
+        self.camera.translation.value = (-1, 1)
+        self.assertEqual(self.camera.resolution.value[::-1], im0.shape[:2])
+
+        # Move a little bit => should have a different image
+        self.camera.updateMetadata({model.MD_POS: (100e-6, 200e-6)})
+        im_move1 = self.camera.data.get()
+        # Note: images are always different, due to synthetic noise
+        test.assert_array_not_equal(im0, im_move1)
```

Comment:
Using MeasureShift it would be possible to verify the amount of movement of the stage.
But probably you thought about that and that is not the goal of this unit test.

Reply:
Looks good!